api/secret.go

```diff
@@ -113,69 +146,19 @@ func (s *Secret) TokenPolicies() ([]string, error) {
 		return s.Auth.Policies, nil
 	}
 
-	if s.Data == nil || s.Data["policies"] == nil {
+	if s.Data == nil {
```

Comment:
Looks like you could get rid of the `if` altogether, since the test is already present in extractListOfStrings.

Reply:
Done